js.go

```diff
@@ -1433,7 +1337,7 @@ func (sub *Subscription) resetOrderedConsumer(sseq uint64) {
 	osid := sub.applyNewSID()
 
 	// Grab new inbox.
-	newDeliver := NewInbox()
+	newDeliver := nc.newInbox()
```

Comment:
@derekcollison Unrelated to this PR, but while I was changing references to NewInbox() in subscribe, I found this one and changed it too. If it should not be changed, let me know.